dxr/plugins/python/tests/test_parse_errors.py

```diff
@@ -0,0 +1,70 @@
+import warnings
+from textwrap import dedent
+
+from dxr.plugins.python.tests import PythonSingleFileTestCase
+
+
+class WarningTestCase(PythonSingleFileTestCase):
+    @classmethod
+    def get_config(cls, config_dir_path):
+        # We need to disable workers because catch_warnings does not
+        # support multi-threaded or multi-process capturing.
+        config = super(WarningTestCase, cls).get_config(config_dir_path)
+        config['DXR']['workers'] = 0
+        return config
+
+    @classmethod
+    def setup_class(cls):
+        # Wrap the normal setup to catch warnings.
+        with warnings.catch_warnings(record=True) as caught_warnings:
+            warnings.simplefilter('always')
+            super(WarningTestCase, cls).setup_class()
+            cls.warnings = caught_warnings
+
+
+class InvalidPythonTest(WarningTestCase):
+    source = dedent("""
+    from foo import bar
+
+    biff = -8bazinvalid  # That ain't right
+
+    class TestClass(object):
+        pass
+    """)
+
+    def test_parse_error(self):
+        """Make sure nothing is indexed for a file with invalid Python
+        code, and that the build process raised warnings about it.
+
+        """
+        self.found_nothing('class:TestClass')
+
+        # One warning for the pre-index analyzing, and one for indexing.
+        self.assertEqual(len(self.warnings), 1)
+        self.assertTrue('Failed to analyze main.py' in
+                        unicode(self.warnings[0].message))
+
+
+class InvalidTokenTest(WarningTestCase):
+    source = dedent("""
+    from foo import bar
+
+    class TestClass(object):
+        pass
+
+    '''Tokenizer fails on un-closed strings. Let's make sure if we do
```

Comment:
It isn't, but it used to be, and I figure since the code will move around a bunch in the future too it's reasonable to test this case in case someday we run the tokenizer first after a refactor.